plutus-core/untyped-plutus-core/test/Spec.hs

```diff
@@ -22,7 +24,9 @@ import UntypedPlutusCore qualified as UPLC
 import Flat
 
 main :: IO ()
-main = defaultMain $ testGroup "Untyped Plutus Core"
+main = do
+  setLocaleEncoding utf8
```

Comment:
What's this for?  I'm sure there's a good reason for it and I'm wondering if we should be doing the same elsewhere.

Reply:
Without this, you get crashes due to mismatched encodings in (at least my) `nix-shell`.

Reply:
To elaborate further: It is assumed that the locale is UTF-8 without setting it to UTF-8 explicitly, meaning that `base` infers the locale from the environment, and this often fails in the context of Nix due to `LANG`, etc. possibly not being set IIRC. This is good practice in general really and should just be the default IMO.

Reply:
I see what you mean: if you set `LC_ALL` to something nonsensical then our current tests start failing with messages like `plutus-core-test: <stdout>: commitBuffer: invalid argument (invalid character)`.  That's useful to know. Thanks!